src/project_types/permissions.rs

```diff
@@ -1,94 +1,56 @@
-
 use std::path::Path;
+use std::ascii::AsciiExt;
 
 use error::Error;
 use utils;
 
 
 #[derive(Clone, Debug, Eq, PartialEq, RustcEncodable, RustcDecodable)]
-pub enum PermissionEnum {
+pub enum Permission {
     Administrate,
-    EditSeq,
-    EditSeqSec,
-}
-
-#[derive(Clone, Debug, Eq, PartialEq, RustcEncodable, RustcDecodable)]
-pub struct Permission {
-    pub which: PermissionEnum,
-    pub target: Option<String>,
+    EditSeq(String),
+    EditSeqSec(String, u32),
 }
 
 impl Permission {
-    /// Creates a new Permission, joining a permission type with a target
-    /// Returns an error if the target is invalid
-    pub fn new(which_enum: PermissionEnum, t: Option<String>) -> Result<Permission, Error> {
-        // Make sure the target is valid for the given permission type
-        try!(Permission::validate_permission(&which_enum, &t));
-
-        // Create permission if valid
-        Ok(Permission {
-            which: which_enum,
-            target: t,
-        })
-    }
-
-    /// Validates the target for the given permission type
-    /// Returns error if invalid target
-    fn validate_permission(permission: &PermissionEnum, target: &Option<String>) -> Result<(), Error> {
-        
-        let valid = match permission {
-            &PermissionEnum::Administrate => {
-                target == &None::<String>
-            },
-            &PermissionEnum::EditSeq => {
-                if target.is_none() {
-                    false
+    /// Creates a new Permission
+    /// Assumes target options are from Docopt, and are therefore safe to unwrap
+    /// if the permission name is correct
+    pub fn new(
+        perm_name: &str,
+        target_sequence: Option<String>,
+        target_section: Option<u32>
+    ) -> Result<Permission, Error> {
+
+        match perm_name.to_ascii_lowercase().as_ref() {
+            "administrate" => Ok(Permission::Administrate),
+            "editseq" => {
+                let sequence_name = target_sequence.unwrap();
+                let project = try!(utils::read_protonfile(None::<&Path>));
+                if project.find_sequence_by_name(&sequence_name).is_none() {
```

Comment:
I prefer to have the errors first. That way the error messages/types are right next to the check that triggered them. It's stylistic preference, so I think keep it like this unless you give me a better reason to switch it.